recipes/openldap/all/conanfile.py

```diff
@@ -1,114 +1,90 @@
 import os
 from conans import ConanFile, AutoToolsBuildEnvironment, tools
 
+required_conan_version = ">=1.43.0"
+
 
 class OpenldapConan(ConanFile):
     name = "openldap"
     description = "OpenLDAP C++ library"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://www.openldap.org/"
-    license = "OpenLDAP"
-    topics = "openldap"
+    license = "OLDAP-2.8"
+    topics = ("ldap", "load-balancer", "directory-access")
     exports_sources = ["patches/*"]
-    settings = {
-        "os": ["Linux"],
-        "compiler": {
-            "gcc": {"version": ["8", "8.2", "10", "10.2"]}
-        },
-        "build_type": ["Debug", "Release"],
-        "arch": ["x86_64"]
-    }
-    generators = "cmake"
+    settings = settings = "os", "compiler", "build_type", "arch"
     options = {
         "shared": [True, False],
-        "cyrus_sasl": [True, False]
+        "fPIC": [True, False],
+        "with_cyrus_sasl": [True, False]
     }
     default_options = {
         "shared": False,
-        "cyrus_sasl": True
+        "fPIC": True,
+        "with_cyrus_sasl": True
+
     }
+    _autotools = None
+    _configure_vars = None
 
     @property
     def _source_subfolder(self):
         return "source_subfolder"
 
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
     def requirements(self):
-        self.requires("openssl/1.1.1k")
-        if self.options.cyrus_sasl:
+        self.requires("openssl/1.1.1m")
+        if self.options.with_cyrus_sasl:
             self.requires("cyrus-sasl/2.1.27")
 
-    def source(self):
-        conan_data = self.conan_data["sources"][self.version]
-        tools.get(url=conan_data["url"], sha256=conan_data["sha256"])
-        extracted_dir = self.name + "-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration(f"{self.name} is only supported on Linux")
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        yes_no = lambda v: "yes" if v else "no"
+        self._autotools = AutoToolsBuildEnvironment(self)
+        configure_args = [
+            "--enable-shared={}".format(yes_no(self.options.shared)),
+            "--enable-static={}".format(yes_no(not self.options.shared)),
+            "--with-cyrus_sasl={}".format(yes_no(self.options.with_cyrus_sasl)),
+            "--with-pic={}".format(yes_no(self.options.get_safe("fPIC", True))),
+            "--without-fetch",
+            "--with-tls=openssl",
+            "--enable-auditlog"]
+        self._configure_vars  = self._autotools.vars
+        self._configure_vars["systemdsystemunitdir"] = os.path.join(self.package_folder, "res")
```

Comment:
The LD_LIBRARY_PATH and LIBS are removed ?

Reply:
injected automatically by Conan